lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -62,75 +66,169 @@ def __init__(self, config):
-        self.validate_params(params)
-
-        # 2) construct the input directory staging area
-        input_directory = self.stage_input(params)
-
-        # 3) update default params
-        self.default_params.update(params)
-        params = self.default_params
-        self.generate_parents = params.get('generate_missing_genes')
-        self.generate_ids = params.get('generate_ids_if_needed')
-        if params.get('genetic_code'):
-            self.code_table = params['genetic_code']
-
-        # 4) Do the upload
-        files = self._find_input_files(input_directory)
-        consolidated_file = self._join_files_skip_empty_lines(files)
-        genome = self.parse_genbank(consolidated_file, params)
-        if params.get('genetic_code'):
-            genome["genetic_code"] = params['genetic_code']
-
-        result = self.gi.save_one_genome({
-            'workspace': params['workspace_name'],
-            'name': params['genome_name'],
-            'data': genome,
-            "meta": params['metadata'],
-        })
-        ref = f"{result['info'][6]}/{result['info'][0]}/{result['info'][4]}"
-        logging.info(f"Genome saved to {ref}")
-
-        # 5) clear the temp directory
-        shutil.rmtree(input_directory)
-
-        # 6) return the result
-        info = result['info']
-        details = {
-            'genome_ref': ref,
-            'genome_info': info
-        }
+    def import_genbank(self, params):
+        print('validating parameters')
+        mass_params = self._set_up_single_params(params)
+        return self._import_genbank_mass(mass_params)[0]
+
+    def import_genbank_mass(self, params):
+        print('validating parameters')
+        self._validate_mass_params(params)
+        return self._import_genbank_mass(params)
+
+    def _set_up_single_params(self, params):
+        # avoid side effects and keep variables in params unmodfied
+        inputs = dict(params)
+        self._validate_params(inputs)
+        ws_id = self._get_int(inputs.pop(_WSID, None), _WSID)
+        ws_name = inputs.pop(_WSNAME, None)
+        if (bool(ws_id) == bool(ws_name)):  # xnor
+            raise ValueError(f"Exactly one of a '{_WSID}' or a '{_WSNAME}' parameter must be provided")
+        if not ws_id:
+            print(f"Translating workspace name {ws_name} to a workspace ID. Prefer submitting "
+                  + "a workspace ID over a mutable workspace name that may cause race conditions")
+            ws_id = self.dfu.ws_name_to_id(ws_name)
+        mass_params = {_WSID: ws_id, _INPUTS: [inputs]}
+        return mass_params
+
+    def _validate_mass_params(self, params):
+        ws_id = self._get_int(params.get(_WSID), _WSID)
+        if not ws_id:
+            raise ValueError(f"{_WSID} is required")
+        inputs = params.get(_INPUTS)
+        if not inputs or type(inputs) is not list:
+            raise ValueError(f"{_INPUTS} field is required and must be a non-empty list")
```

Comment:
Should probably check that there are < 10k inputs or the workspace will reject it

Reply:
May I make this as TODO?

Similar to AssemblyUtil, I will add them in parallelization PR
https://github.com/kbaseapps/AssemblyUtil/blob/c40f2358ab0a17683aac8ebbdec40464299b1e47/lib/AssemblyUtil/FastaToAssembly.py#L38

Reply:
Eh, it's 2 lines, I'd just do it personally, but if you want to do it in a later PR that's fine

Reply:
ah, is not 2 lines. Will do it in a later PR.
https://github.com/kbaseapps/AssemblyUtil/blob/c40f2358ab0a17683aac8ebbdec40464299b1e47/lib/AssemblyUtil/FastaToAssembly.py#L330

Reply:
Oh, I was talking about just throwing an error, but yeah, for batching that's a lot more

Reply:
Will fix in a later PR

Reply:
:+1: